### PR TITLE
chore: Upgrade kubectl to version 1.30.0

### PR DESCRIPTION
**Pull request by SISheogorath**
Just a regular upgrade, since 1.25.4 is a bit dated by now. Also making sure we actually used the docker.io image unless explicitly changed.

### CODE REVIEW

**Comment by IT-Cru** [comment]
Does this currently also blocks upgrades to k3s 1.30.x? If yes, would be nice to get it merged :)

**Comment by SISheogorath** [comment]
I don't think it should have that impact, it's just updates for good measure. But you can validate it the impact on your cluster, if you want. All you need to do is adjusting the environment variable as shown in the manifest.

**Comment by brandond** [comment]
kubectl is widely compatible with both newer and older versions of the apiserver. Upgrading or not upgrading doesn't block or unblock anything.

File: manifests/system-upgrade-controller.yaml
Comment (brandond) [inline manifests/system-upgrade-controller.yaml:25]:
```suggestion
  SYSTEM_UPGRADE_JOB_KUBECTL_IMAGE: "docker.io/rancher/kubectl:v1.30.4"
```

File: pkg/upgrade/job/job.go
Comment (brandond) [inline pkg/upgrade/job/job.go:27]:
```suggestion
	defaultKubectlImage            = "docker.io/rancher/kubectl:v1.30.4"
```

**Review by brandond — CHANGES_REQUESTED** [review]
lgtm, but lets bump this to latest before merging

**Comment by brandond** [comment]
nm, we actually already bumped these in https://github.com/rancher/system-upgrade-controller/pull/326

**Pull request closed without merging**
